cms/djangoapps/import_from_modulestore/models.py

```diff
@@ -0,0 +1,172 @@
+
+    def __str__(self):
+        return f'{self.source_key} - {self.target}'
+
+    def ready(self) -> None:
+        """
+        Set import status to ready.
+        """
+        self.status = ImportStatus.READY
+        self.save()
+
+    def imported(self) -> None:
+        """
+        Set import status to imported and clean related staged content.
+        """
+        self.status = ImportStatus.IMPORTED
+        self.save()
+        self.clean_related_staged_content()
+
+    def cancel(self) -> None:
+        """
+        Cancel import action and delete related staged content.
+        """
+        self.status = ImportStatus.CANCELED
+        self.save()
+        self.clean_related_staged_content()
+
+    def error(self) -> None:
+        """
+        Set import status to error and clean related staged
+        """
+        self.status = ImportStatus.ERROR
+        self.save()
+        self.clean_related_staged_content()
+
+    def clean_related_staged_content(self) -> None:
+        """
+        Clean related staged content.
+        """
+        for staged_content_for_import in self.staged_content_for_import.all():
+            staged_content_for_import.staged_content.delete()
+
+    class Meta:
+        verbose_name = _('Import from modulestore')
+        verbose_name_plural = _('Imports from modulestore')
+
+    @classmethod
+    def get_by_uuid(cls, import_uuid: str) -> Self | None:
+        """
+        Get an import task by its ID.
+        """
+        return cls.objects.filter(uuid=import_uuid).first()
+
+    @classmethod
+    def get_ready_by_uuid(cls, import_uuid: str) -> Self | None:
+        """
+        Get an import task by its UUID.
+        """
+        return cls.objects.filter(uuid=import_uuid, status=ImportStatus.READY).first()
+
+    def get_staged_content_by_block_usage_id(self, block_usage_id: str) -> Optional["StagedContent"]:
+        """
+        Get staged content by block usage ID.
+        """
+        staged_content_for_import = self.staged_content_for_import.filter(
+            staged_content__tags__icontains=block_usage_id
+        ).first()
+        return getattr(staged_content_for_import, 'staged_content', None)
+
+
+class PublishableEntityMapping(TimeStampedModel):
+    """
+    Represents a mapping between a source usage key and a target publishable entity.
+    """
+
+    source_usage_key = UsageKeyField(
+        max_length=255,
+        help_text=_('Original usage key/ID of the thing that has been imported.'),
+    )
+    target_package = models.ForeignKey(LearningPackage, on_delete=models.CASCADE)
+    target_entity = models.ForeignKey(PublishableEntity, on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ('source_usage_key', 'target_package')
+
+    def __str__(self):
+        return f'{self.source_usage_key} - {self.target_entity}'
+
+
+class PublishableEntityImport(TimeStampedModel):
+    """
+    Represents a container version that has been imported into a content library.
```

Comment:
```suggestion
    Represents a publishableentity version that has been imported into a learning package (e.g. content library)
```
docstring is outdated

Reply:
Updated.